packages/react-dom/src/client/ReactInputSelection.js

```diff
@@ -26,7 +26,8 @@ export function hasSelectionCapabilities(elem) {
   const nodeName = elem && elem.nodeName && elem.nodeName.toLowerCase();
   return (
     nodeName &&
-    ((nodeName === 'input' && elem.type === 'text') ||
+    ((nodeName === 'input' &&
+      ['text', 'email', 'tel'].indexOf(elem.type) >= 0) ||
```

Comment:
Instead of defining an inline array, let's just add some additional cases to the boolean expression. 

```js
(
nodeName === 'input' &&
(elem.type === 'text' || elem.type === 'email' || elem.type === 'tel')
)
```